cli/src/main.rs

```diff
@@ -17,27 +22,91 @@ enum Command {
         /// The path to the supegraph schema file, or `-` for stdin
         supegraph_schema: PathBuf,
     },
+    /// Outputs the query graph from a supergraph schema or subgraph schemas
+    QueryGraph {
+        /// Path(s) to one supergraph schema file or multiple subgraph schemas.
+        schemas: Vec<PathBuf>,
+    },
+    /// Outputs the federated query graph from a supergraph schema or subgraph schemas
+    FederatedGraph {
+        /// Path(s) to one supergraph schema file or multiple subgraph schemas.
+        schemas: Vec<PathBuf>,
+    },
 }
 
-fn main() {
+fn main() -> ExitCode {
     let args = Args::parse();
-    match args.command {
-        Command::Api { supegraph_schema } => to_api_schema(supegraph_schema),
+    let result = match args.command {
+        Command::Api { supegraph_schema } => to_api_schema(&supegraph_schema),
+        Command::QueryGraph { schemas } => dot_query_graph(&schemas),
+        Command::FederatedGraph { schemas } => dot_federated_graph(&schemas),
+    };
+    match result {
+        Err(error) => {
+            eprintln!("{error}");
+            ExitCode::FAILURE
+        }
+
+        Ok(_) => ExitCode::SUCCESS,
     }
 }
 
-fn to_api_schema(input_path: PathBuf) {
+fn to_api_schema(input_path: &PathBuf) -> Result<(), FederationError> {
     let input = if input_path == std::path::Path::new("-") {
         io::read_to_string(io::stdin()).unwrap()
     } else {
         fs::read_to_string(input_path).unwrap()
     };
-    let supergraph = apollo_federation::Supergraph::new(&input).unwrap();
-    let api_schema = supergraph
-        .to_api_schema(apollo_federation::ApiSchemaOptions {
-            include_defer: true,
-            include_stream: false,
-        })
-        .unwrap();
-    println!("{}", api_schema.schema())
+    let supergraph = apollo_federation::Supergraph::new(&input)?;
+    let api_schema = supergraph.to_api_schema(apollo_federation::ApiSchemaOptions {
+        include_defer: true,
+        include_stream: false,
+    })?;
+    println!("{}", api_schema.schema());
+    Ok(())
+}
+
+/// Load either single supergraph schema file or compose one from multiple subgraph files.
+fn load_supergraph(
+    file_paths: &[PathBuf],
+) -> Result<apollo_federation::Supergraph, FederationError> {
+    if file_paths.is_empty() {
+        panic!("Error: missing command arguments");
+    } else if file_paths.len() == 1 {
+        let doc_str = std::fs::read_to_string(&file_paths[0]).unwrap();
+        apollo_federation::Supergraph::new(&doc_str)
+    } else {
+        let schemas: Vec<_> = file_paths
+            .iter()
+            .map(|pathname| {
+                let doc_str = std::fs::read_to_string(pathname).unwrap();
+                let url = "file://".to_string() + pathname.to_str().unwrap();
```

Comment:
Ah, missed this the first time, it's almost always better to use the `format!()` macro than the string `+` operator, it reads a bit better and requires fewer memory allocations.
```suggestion
                let url = format!("file://{}", pathname.to_str().unwrap());
```